File: leet-code/0344-reverseString.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function (s) {
  let leftIndex = 0;
  let rightIndex = s.length - 1;

  while (leftIndex < rightIndex) {
    let temp = s[leftIndex];
    s[leftIndex] = s[rightIndex];
    s[rightIndex] = temp;

    leftIndex++;
    rightIndex--;
  }
};

File: leetcode/0344-reverse-string.py
def reverseString(s):
    """
    Do not return anything, modify s in-place instead.
    """
    left_index = 0
    right_index = len(s) - 1

    while (left_index < right_index):
        s[left_index], s[right_index] = s[right_index], s[left_index]

        left_index += 1
        right_index -= 1
